hls4ml/model/optimizer/passes/bn_quant.py

```diff
@@ -16,35 +16,12 @@ def initialize(self):
         inp = self.get_input_variable()
         shape = inp.shape
         dims = inp.dim_names
-        precision = self.model.config.backend.convert_precision_string(inp.type.precision)
-        W, I, F = precision.width, precision.integer, precision.fractional
-        original_name = self.attributes.get('original_name')
-        variance = self.model.get_weights_data(original_name, 'moving_variance')
-        mean = self.model.get_weights_data(original_name, 'moving_mean')
-        gamma = self.model.get_weights_data(original_name, 'gamma')
-        beta = self.model.get_weights_data(original_name, 'beta')
-        mean_quantizer = self.get_attr('mean_quantizer')
-        variance_quantizer = self.get_attr('variance_quantizer')
-        gamma_quantizer = self.get_attr('gamma_quantizer')
-        beta_quantizer = self.get_attr('beta_quantizer')
-        mean = mean_quantizer(mean) if mean_quantizer is not None else mean
-        variance = variance_quantizer(variance) if variance_quantizer is not None else variance
-        gamma = gamma_quantizer(gamma) if gamma_quantizer is not None else gamma
-        beta = beta_quantizer(beta) if beta_quantizer is not None else beta
-        epsilon = self.attributes.get('epsilon')
-        threshold = mean - beta * np.sqrt(variance + epsilon) / gamma
         if self.get_attr('quantize') == 2:
             self.add_output_variable(shape, dims, precision=XnorPrecisionType())
-            threshold = np.floor(threshold * 2**F) / 2**F
-            self.add_weights_variable(name='threshold', var_name='t{index}', data=threshold, type_name='threshold{index}_t', precision=inp.type.precision)
         elif self.get_attr('quantize') == 3:
             self.add_output_variable(shape, dims, precision=IntegerPrecisionType(width=2))
-            threshold_hi = 0.5/(gamma/np.sqrt(variance + epsilon)) + threshold
-            threshold_lo = -0.5/(gamma/np.sqrt(variance + epsilon)) + threshold
-            threshold_hi = np.floor(threshold_hi * 2**F) / 2**F
-            threshold_lo = np.floor(threshold_lo * 2**F) / 2**F
-            self.add_weights_variable(name='threshold_hi', var_name='th{index}', data=threshold_hi, type_name='threshold_hi_{index}_t', precision=inp.type.precision)
-            self.add_weights_variable(name='threshold_lo', var_name='tl{index}', data=threshold_lo, type_name='threshold_lo_{index}_t', precision=inp.type.precision)
+        else:
+            print("Not adding output variable")
```

Comment:
Is this even reachable? If so, maybe a more descriptive message would help with debugging and an exception should be raised since we cannot recover from this.

Reply:
I've replaced the `print` with a `raise Exception`